src/build.js

```diff
@@ -0,0 +1,43 @@
+/* @flow */
+import path from 'path';
+import {createWriteStream} from 'fs';
+import streamToPromise from 'stream-to-promise';
+
+import * as fs from './util/promised-fs';
+import {zipDir} from './util/zip-dir';
+
+
+export default function build({sourceDir, buildDir}: Object) {
```

Comment:
It's early but there is an [ES8 proposal](https://github.com/sirisian/ecmascript-types) (which flow seems to follow) for type annotations. FWIW.

> https://www.npmjs.com/package/type-check

This is a runtime check which I'd like to avoid because it has performance implications. Flow is completely static. The first babel transform just strips out the annotations.

I definitely think typescript is a bad way to go because it forces you to type everything. I like how Flow is opt-in and how it supports duck-typed interfaces (like Go). I personally don't find the annotations hard to read. I think that for anyone who has even rudimentary knowledge of another static language (Go, Scala, Java, Objective C, ...), it would be intuitive to figure out Flow's annotations. Flow's docs are pretty good although I find their layout confusing. 

I hear what you're saying about hindering contributions though. It will slow someone down the first time if they've never seen Flow. Ultimately, I think it could make for higher quality contributions -- we could rely on interfaces to throw usage errors and it should be easier for a novice to write unit tests because they won't have to "know" too much about the program structure. That is, they will be yelled at by the compiler which should help them figure out the structure

> Don't get me wrong; I see the problem - I'm just not sure this is a good solution.

I'm open to other ideas for solutions. I don't think it would be hard to strip out Flow annotations if we had to. It only requires you to annotate exported functions (inputs and outputs) so I've been trying to do no more than that.